### PR TITLE
Detect golang version in integration-cli

### PR DESCRIPTION
**Pull request by LK4D4**
Also changed some tests which depends on golang version

### CODE REVIEW

File: integration-cli/docker_cli_import_test.go [hunk 1]
```diff
@@ -22,8 +22,15 @@ func TestImportDisplay(t *testing.T) {
...
-		t.Fatalf("display is messed up: %d '\\n' instead of 2", n)
+	// TODO: remove check after drop of go1.2 support
+	var expected int
+	if strings.HasPrefix(goVersion, "go1.2") {
+		expected = 2
+	} else {
+		expected = 3
+	}
+	if n := strings.Count(out, "\n"); n != expected {
+		t.Fatalf("display is messed up: %d '\\n' instead of %d, go version: %s", n, expected, goVersion)
```
Comment (tianon) [inline integration-cli/docker_cli_import_test.go]: I'm not so sure this is correct -- why does Go 1.2 generate three lines instead of two?  What are they?

Comment (LK4D4) [inline integration-cli/docker_cli_import_test.go]: There is additional line at the end of input. I have no slightest idea from where. We tried to figure out but failed. Seems like some change in `http` package changed behavior of our json streams.

File: integration-cli/docker_test_vars.go [hunk 2]
```diff
@@ -23,6 +26,8 @@ var (
 	containerStoragePath = dockerBasePath + "/containers"
 
 	workingDirectory string
+
+	goVersion string
```
Comment (ahmetalpbalkan) [inline integration-cli/docker_test_vars.go]: I think `daemonGoVersion` is a better name. at first I thought if this is `runtime.Version()` of the cli or not.

Comment (LK4D4) [inline integration-cli/docker_test_vars.go]: Yup, you're right.

**Comment by LK4D4** [comment]
@ahmetalpbalkan I changed variable name.

**Comment by jfrazelle** [comment]
I think we need to get to the heart of the matter instead

**Comment by LK4D4** [comment]
@jfrazelle What if there is no heart of matter? Then we just waste time on thing which used by not a single person in a world.

**Comment by jfrazelle** [comment]
lol idk I was more hoping some of ahmets patches would magically fix who
knows

**Comment by LK4D4** [comment]
@jfrazelle :) I think @ahmetalpbalkan is not working on cross-version support. But he can!

**Comment by tianon** [comment]
@vbatts you're the one I know of who has cared about aggressive Go
retrocompatibility previously -- will you survive if we drop Go 1.2 support
and call this case closed? :smile:

**Comment by LK4D4** [comment]
And go1.3, why not? :)

**Comment by tianon** [comment]
IMO, Go 1.3 support is still worth keeping for a little while longer,
especially since we've got all the tests passing (ie, it's currently
working fine).

**Comment by vbatts** [comment]
@tianon now that gcc5 is frozen and hoped to release imminently, I am less strong on go1.2 compat. (As gcc5 will have a gcc-go that support go1.4)

@LK4D4 let's keep go1.3 for a while ...

**Comment by jfrazelle** [comment]
meh go 1.2

**Comment by jfrazelle** [comment]
<3

**Pull request closed without merging**